controllers/nstemplatetier/nstemplatetier_controller_test.go

```diff
@@ -77,6 +83,256 @@ func TestReconcile(t *testing.T) {
+				r, req, cl := prepareReconcile(t, base1nsTierWithRevisions.Name, append(tierTemplates, base1nsTierWithRevisions)...)
+				// when
+				// check no TTR is present before reconciling
+				tiertemplaterevision.AssertThatTTRs(t, cl, base1nsTierWithRevisions.GetNamespace()).DoNotExist()
+				_, err = r.Reconcile(context.TODO(), req)
+				// then
+				require.NoError(t, err)
+				// check that revisions field was populated
+				tiertest.AssertThatNSTemplateTier(t, "base1ns", cl).
+					HasStatusTierTemplateRevisions(tierTemplatesRefs)
+				// check that expected TierTemplateRevision CRs were created
+				ttrs := toolchainv1alpha1.TierTemplateRevisionList{}
+				err = cl.List(context.TODO(), &ttrs, runtimeclient.InNamespace(base1nsTierWithRevisions.GetNamespace()))
+				require.NoError(t, err)
+				require.Len(t, ttrs.Items, len(tierTemplatesRefs))
+
+			})
+
+			t.Run("TTR name should stay within 63 chars, so that they can be used as labels", func(t *testing.T) {
+				// given
+				// the TierTemplateRevision CRs are missing, and their name are based on the tier name.
+				// Making the TierName already 63chars long we test that the TTR name stays within 63 chars
+				veryLongTierName := "somerandomstringtomakethenamelongerthan63chars12345678912345678"
+				tierWithVeryLongName := tiertest.Tier(t, veryLongTierName, tiertest.NSTemplateTierSpecWithTierName(veryLongTierName),
+					tiertest.WithParameter("DEPLOYMENT_QUOTA", "60"),
+				)
+				tierHash, err := hash.ComputeHashForNSTemplateTier(tierWithVeryLongName)
+				require.NoError(t, err)
+				// let's set the update so that we can skip the first reconciliation loop when this is updated
+				tierWithVeryLongName.Status.Updates = []toolchainv1alpha1.NSTemplateTierHistory{
+					{
+						StartTime: metav1.Now(),
+						Hash:      tierHash,
+					},
+				}
+				tierTemplatesWithLongNames := initTierTemplates(t, withTemplateObjects(crq), tierWithVeryLongName.Name)
+				r, req, cl := prepareReconcile(t, tierWithVeryLongName.Name, append(tierTemplatesWithLongNames, tierWithVeryLongName)...)
+				// when
+				// check no TTR is present before reconciling
+				tiertemplaterevision.AssertThatTTRs(t, cl, tierWithVeryLongName.GetNamespace()).DoNotExist()
+				_, err = r.Reconcile(context.TODO(), req)
+				// then
+				require.NoError(t, err)
+				// check that expected TierTemplateRevision CRs were created
+				// with the expected length
+				ttrs := toolchainv1alpha1.TierTemplateRevisionList{}
+				err = cl.List(context.TODO(), &ttrs, runtimeclient.InNamespace(tierWithVeryLongName.GetNamespace()))
+				require.NoError(t, err)
+				require.Len(t, ttrs.Items, 7)
+				for _, ttr := range ttrs.Items {
+					assert.Empty(t, validation.IsDNS1123Label(ttr.GetName()))
+				}
```

Comment:
couldn't you use the `ForEach` assertion for this as well?
```go
tiertemplaterevision.AssertThatTTRs(t, cl, tierWithVeryLongName.GetNamespace()).
	ForEach(func(ttr *toolchainv1alpha1.TierTemplateRevision) {
		assert.Empty(t, validation.IsDNS1123Label(ttr.GetName()))
	})
```
and if you add another assertion verifying also the number of CRs, then you can do:
```go
tiertemplaterevision.AssertThatTTRs(t, cl, tierWithVeryLongName.GetNamespace()).
	NumberOfPresentCRs(7).
	ForEach(func(ttr *toolchainv1alpha1.TierTemplateRevision) {
		assert.Empty(t, validation.IsDNS1123Label(ttr.GetName()))
	})
```

Reply:
btw, I'm wondering why it didn't fail before the actual logic was modified https://github.com/codeready-toolchain/host-operator/pull/1103#discussion_r1913408493

Reply:
> btw, I'm wondering why it didn't fail before the actual logic was modified https://github.com/codeready-toolchain/host-operator/pull/1103#discussion_r1913408493

Unless I'm drunk , It looks like the `Create` adjusts the length automatically to 63chars 😅  ( so no need to handle it in the controller 🤦‍♂️ ) :
![image](https://github.com/user-attachments/assets/19a31fb4-19c6-4167-a7ba-62e7e6d32e36)

Reply:
In fact if I drop the check from the controller:
```
	maxNameLength := validation.DNS1123LabelMaxLength - 6
	if len(newTTRName) > maxNameLength {
		newTTRName = newTTRName[0:maxNameLength]
	}
```
the tests are still passing.

Reply:
implemented your suggestion in the test and dropped the DNS compliancy check from the controller in https://github.com/codeready-toolchain/host-operator/pull/1103/commits/9c2769b9924d15fe09920d3ae3d4a9a59a14c704

Reply:
yup, k8s automatically shortens the length of the name https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/names/generate.go